src/aimes/emgr/strategy.py

```diff
@@ -114,7 +115,74 @@ def derive_execution_stategy_skeleton(cfg, workload, resources, run):
 
     return strategy
 
+# -----------------------------------------------------------------------------
+def derive_execution_strategy_skeleton_osg(cfg, workload, resources, run):
+
+    #   Initializing data structures
+    strategy = {}
+    strategy['heuristic'] = {}
+    strategy['inference'] = {}
+
+    #   Defining level of concurrency (in %)
+    strategy['heuristic']['percentage_concurrency'] = cfg['pct_concurrency']
+
+    #   Defining the percentage of available resources to use
+    strategy['heuristic']['percentage_resources'] = cfg['pct_resources']
+
+    #   Initializing list of resources to use
+    strategy['inference']['target_resources'] = list()
+
+    #   Defining resources to use for resources which do and do not support bundles
+    for i in range(cfg['num_osg_pilots']):
+        strategy['inference']['target_resources'].append('osg.xsede-virt-clust')
+
+        #   If early binding, then there's no point in adding more than one resources
+        if run['binding'] == 'early':
+                break
+
+    #   Choosing the Number of pilots, assuming max concurrency. Casting the number
+    #   OSG pilots to an int is to ensure that the value of the pilots specified
+    #   is an int. Assertion that the number of pilots submitted is also to ensure
+    #   the user will submit a reasonable number of pilots
+    if strategy['heuristic']['percentage_resources'] == 100:
+        strategy['inference']['number_pilots'] = max(int(cfg['num_osg_pilots']), 1)
+        assert strategy['inference']['number_pilots'] >= 1, "Must specify a positive integer for the number of OSG pilots to submit\n"
+            
+    #   Choosing the scheduler for the CUs. If there is only one pilot. then
+    #   submit directly to the pilot. If there is more than one pilot. The number
+    #   of pilots has been asserted to be a strictly positive integer
+    if strategy['inference']['number_pilots'] > 1:
+        strategy['inference']['rp_scheduler'] = 'SCHED_BACKFILLING'
+    else:
+        strategy['inference']['rp_scheduler'] = 'SCHED_DIRECT_SUBMISSION'
```

Comment:
I am fine with either Direct Submission or Round Robin. If we choose to use Round Robin instead, we will need to make the corresponding changes in XSEDE-equivalent of the function